site/content/docs/contributing/1.0-roadmap.md

```diff
@@ -15,25 +15,41 @@ as a project.
 
 High level goals for kind include:
 
-- [ ] Reach [1.0] status / 
-stability by the end of Q1 2019 / the Kubernetes 1.14 release cycle
+## Beta Requirements
+
+To reach "beta" [grade][deprecation-policy] kind needs to at minimum:
+
+- [x] Improve documentation (Though this will eternally be "In Progress" !)
+  - [X] create a documentation site - [#268]
+  - [ ] expand examples of using kind
+  - [x] cover known issues, debugging, work-arounds, etc.
+- [x] Reliably [pass][kind-conformance-dashboard] the Kubernetes [conformance tests]
+ - [x] [Certify][certified] Conformance
 - [x] Support multi-node clusters - [#117]
-- [ ] Support offline / air-gapped clusters
-  - [ ] pre-loaded / offline CNI - [#200]
-- [ ] Support mounting host directories - [#62]
-- [ ] Improve Windows support
+- [x] Support offline / air-gapped clusters
+  - [x] pre-loaded / offline CNI - [#200]
+- [x] Support mounting host directories - [#62]
+- [x] Improve Windows support
   - [x] add Windows binaries to releases - [#155]
-  - [ ] improve instructions for KUBECONFIG in particular
-- [ ] Improve documentation - In Progress ...
-  - [ ] expand examples of using kind
-  - [ ] cover known issues, debugging, work-arounds, etc.
-  - [X] create a documentation site - [#268]
-- [ ] Improving logging and debuggability
-  - [ ] enable audit-logging
-- [ ] Support non-amd64 architectures (namely ARM) - [#166]
+  - [x] improve instructions for KUBECONFIG in particular
+- [ ] Support usage as a properly versioned, supported, and documented library 
```

Comment:
@BenTheElder @aojea 

something i wanted to recommend (without logging a ticket) is to not employ the kube-proxy pattern where an alpha API version grows with new features being added without bumping the API version.

in kubeadm we switched away from that somewhere after i joined, yet there were still cases when someone just approved changes on the old API. this is a bad pattern and those who import the kind public types would expect them to not change without a version bump. there are other areas of k8s that are guilty of this - e.g. some `v1` areas that don't want to introduce a `v2alpha1`.

this however means that new features have to be accumulated in the new API version that is not released yet.
e.g. 0.5.0 -> v1alpha4.

Reply:
Even go itself adds new fields to structs I thought, they just must have sane defaults and the type accepted everywhere.

Will consider though.

Reply:
Kubernetes has some very extensive rules for this fwiw, we do our best to follow those.

Reply:
I'm not clear what the ask is here, but I can't tell that it requires any changes to this line.

To me, I think use as a library means we're going to have some specific use cases documented, and those are going to be documented and tested.

We should make sure that we spell out what "properly versioned" means as part of these use cases.

Reply:
I think we need a bit more work into determining what "properly versioned" is, but roughly it should be possible to follow semver without every release being a major version, and we should not violate semver.

That means things like the public API types being stable from a compatibility perspective, and the non-public API types not being part of the library. Changes to the library methods need to be non-breaking, which means moves like ensuring cluster creation takes an extensible set of options so we can add new ones with defaulting without breaking the existing options usage.

Supported is mostly redundantly emphasizing this, IE not breaking people and having a public interface.

documented of course is self explanatory, we have poor / nonexistent godocs.

Reply:
[working on this one]